pytorch_lightning/loops/epoch/evaluation_epoch_loop.py

```diff
@@ -241,3 +245,10 @@ def _should_track_batch_outputs_for_epoch_end(self) -> bool:
         if self.trainer.testing:
             return is_overridden("test_epoch_end", model)
         return is_overridden("validation_epoch_end", model)
+
+    @property
+    def _has_completed_validation_batch(self) -> bool:
+        return self.batch_progress.current.ready == self.batch_progress.current.completed
+
+    def _num_val_batches_reached(self) -> bool:
+        return self._has_completed_validation_batch and self.batch_progress.is_last_batch
```

Comment:
one is a property, the other is a method. plus, they are protected yet never used by the class here. this makes little sense. imo, either

1) make them both public properties
2) move them to the loop that actually uses this information, then they can stay protected there

my preference is 2)

Reply:
@carmocca Any thoughts there ? Should we move them to the TrainingEpochLoop ?

Reply:
Done !

Reply:
@awaelchli one problem of option 2 is that if the child changes (for example if the batch_progress properties are customized) then this code cannot be easily adapted as it is defined in the parent.

Reply:
accessing a protected member of one loop from another offers the same problem. option 1 in my comment above is still  possible I think